sys/kern/vfs_bio.c

```diff
@@ -4888,17 +4888,25 @@ vm_hold_free_pages(struct buf *bp, int newbsize)
 int
 vmapbuf(struct buf *bp, int mapbuf)
 {
+	void * __capability b_data;
 	vm_prot_t prot;
 	int pidx;
+#if __has_feature(capabilities) && !defined(__CHERI_PURE_CAPABILITY__)
+	b_data = cheri_capability_build_user_data(CHERI_CAP_USER_DATA_PERMS,
+	    (__cheri_addr vaddr_t)bp->b_data, bp->b_bufsize, 0);
+	KASSERT((__cheri_addr vaddr_t)b_data == (vaddr_t)bp->b_data,
+	    ("failed to generate capability for bp->b_data == %p", bp->b_data));
```

Comment:
I don't understand the point of this assert. The function, if it returns, always returns a cap with address equal to the requested base+offset.